### ISSUE TITLE
[Enhancement] - Deprovision/Provision - change scope_schema

### ISSUE THREAD

**Issue by mssergiosilva** [Mimetis/Dotmim.Sync#234]
Hi,

First thank you this great sync framework!
This new version is giving me awesome fight getting work =) I have to refactoring all my code to this new version.

If Deprovision a table, change the filter and Provision the scope_schema isn't changed.

To workaround it I have Deprovision/Provision the scope.

Kind regards

**Comment by jeremy-waguet** [comment]
@mssergiosilva The workaround I implemented to this issue is : 

```c#
if (isClentReprovisioning)
            {
                conn.Open();

                // Set scope_schema to NULL for sync_scope_schema to get populated again on next sync
                using (var cmd = conn.CreateCommand())
                {
                    cmd.CommandText = "UPDATE [dbo].[scope_info] SET [sync_scope_schema] = NULL";
                    cmd.ExecuteNonQuery();
                }
            }
```

**Comment by Mimetis** [comment]
This issue will be fixed in the next version `v0.5`
Hopefully, available before end of April :)

**Comment by jeremy-waguet** [comment]
My workaround actually works and I may even not need to reprovision until then.Once again thank for this smazing syncfx replacement (with easier API). Couls you please just updater begginer tutorial about reprovisioning required by (De) Provision Async method does not take into account the new? SyncSet class in your docs. I looked at the API and found that than provider.ReadSchema can do the job by giving the same SyncSetup as when syncing, a connection instance and null for the dbtransaction. Could you please update the donc as there is no mention to SyncSet as of today.Also there is no mention in the docs to SyncProvision.Scope.So it took me a few hours to get the trick.Once again thanks a lot for this framework.On 28 Mar 2020 17:32, Sébastien Pertus <notifications@github.com> wrote:
This issue will be fixed in the next version v0.5
Hopefully, available before end of April :)

**Comment by Mimetis** [comment]
@mssergiosilva The `v0.5` branch is available.

You can test if from nuget. 

https://www.nuget.org/packages/Dotmim.Sync.Core/0.5.0-beta-0299

You can manually install it, through this command line:
``` bash
dotnet add package Dotmim.Sync.Core --version 0.5.0-beta-0299
```

Or through Visual Studio
Be sure to select **Pre-release** to see the packages, inside Visual Studio

![image](https://user-images.githubusercontent.com/4592555/78261943-2e922180-7500-11ea-970c-30a1bc69b528.png)

Or you can directly grab the code from the branch ``Mimetis-v0.5` here : https://github.com/Mimetis/Dotmim.Sync/tree/Mimetis-v0.5
You will have to clone it locally and rebuild it, to use it.
Rebuild:
- `Dotmim.Sync.Core`
- `Dotmim.Sync.MySql`
- `Dotmim.Sync.Sqlite`
- `Dotmim.Sync.SqlServer`
- `Dotmim.Sync.SqlServer.ChangeTracking`
- `Dotmim.Sync.Web.Client`
- `Dotmim.Sync.Web.Server`

**Be careful, there is some breaking changes, and the doc is not up to date with these changes.**

If you want to test it, be sure to start from a **fresh new server and client databases**, because there is no "upgrade path from `v0.4` to `v0.5`" ... *yet*

If you need some example, do not hesitate to search in:
- Tests projects: https://github.com/Mimetis/Dotmim.Sync/tree/Mimetis-v0.5/Tests/Dotmim.Sync.Tests 
- Console sample: https://github.com/Mimetis/Dotmim.Sync/tree/master/Samples/Dotmim.Sync.SampleConsole
- Web Api project: https://github.com/Mimetis/Dotmim.Sync/tree/Mimetis-v0.5/Samples/Dotmim.Sync.SampleWebServer

If you are testing this version, let me know what you find / need
